crates/consensus/executor/src/subscriber.rs

```diff
@@ -309,29 +326,13 @@ impl<DB: Database> Subscriber<DB> {
     ) -> SubscriberResult<ConsensusOutput> {
         let num_blocks = deliver.num_primary_blocks();
         let num_certs = deliver.len();
-
-        // get the execution address of the authority or use zero address
-        let leader = self.inner.committee.authority(deliver.leader.origin());
-        let address = if let Some(authority) = leader {
-            authority.execution_address()
-        } else {
-            error!(target: "subscriber", "Execution address missing for {}", &deliver.leader.origin());
-            return Err(SubscriberError::UnexpectedAuthority(deliver.leader.origin().clone()));
-        };
-
-        let early_finalize = if self.consensus_bus.node_mode().borrow().is_active_cvv() {
-            // We are a CVV so we can finalize early.
-            true
-        } else {
-            // Not a CVV so be more conservative about finalizing blocks.
-            false
-        };
+        // active cvvs always finalize early bc they are the authorities
+        let early_finalize = self.consensus_bus.node_mode().borrow().is_active_cvv();
```

Comment:
I wonder who wrote this "if" originally (think it was me)....  I can only hope there was more to that "if" at one point....  Still no excuse for this, good catch.